beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/hive/HiveClient.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.core.hive;
+
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expediagroup.beekeeper.core.error.BeekeeperException;
+
+import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+
+public class HiveClient {
+
+  private static final Logger log = LoggerFactory.getLogger(HiveClient.class);
+
+  private CloseableMetaStoreClient client;
+  private final boolean dryRunEnabled;
+
+  public HiveClient(CloseableMetaStoreClient client, boolean dryRunEnabled) {
+    this.client = client;
+    this.dryRunEnabled = dryRunEnabled;
+  }
+
+  /**
+   * Will delete the table metadata. Error is not thrown if table not found.
+   * 
+   * @param database
+   * @param tableName
+   */
+  public boolean deleteMetadata(String database, String tableName) {
```

Comment:
Wouldn't `dropTable` be a more accurate method name? Or have you gone with this name to distinguish between deleting the data in the table? I still think I'd prefer the `dropTable` name as the behaviour is more consistent with how Hive works and this is a `HiveClient` class.